### DISCUSSION TITLE
Could you please implement a method for different Percentile, as sampleTimePercentile90?

### DISCUSSION THREAD

**Discussion by almeidas-tatiane** [abstracta/jmeter-java-dsl#182] [Ideas]
I just loved the idea to use an assertion to verify the behavior of the test with the use of sampleTimePercentile99.

I'd like to recomend another percentiles be implemented like 90. Or maybe a generic method sampleTimePercentile where we can type the value in parenthesis.

It will be a big help! Usually in my tests I use percentile 90 as reference.

Thanks,

Tatiane

**Comment by rabelenda** [comment]
Hello, thank you for taking the time to ask this.

Currently the API already provides a method not only to obtain percentile 99, but also 95 and 90. Check [TimeMetricSummary](https://github.com/abstracta/jmeter-java-dsl/blob/master/jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/stats/TimeMetricSummary.java).

Is this enough for your use case?

If not, can you provide examples where you need more flexibility and other percentiles calculations?

Regards
